BookingProject/BookingWebAPI/BookingWebApi.Application/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,124 @@
+            }
+
+            var result = await _signInManager.CheckPasswordSignInAsync(user, loginDto.Password, false);
+
+            if (!result.Succeeded)
+            {
+                return new ApiResponse<NewUserDto>(errorMessages: "Login wasn`t successful", errors: new List<string>() { "Email not found and/or password incorrect" });
+            }
+
+            var newUserDto = new NewUserDto
+            {
+                Username = user.UserName,
+                Email = user.Email,
+                Token = _tokenService.CreateToken(user)
+            };
+
+            return new ApiResponse<NewUserDto>(newUserDto, "Successful login");
+        }
+
+        public async Task<ApiResponse<NewUserDto>> Register(RegisterDto registerDto)
+        {
+            var resultValidator = await _validatorRegisterDto.ValidateAsync(registerDto);
+
+            if (!resultValidator.IsValid)
+            {
+                var errorsValidation = resultValidator.Errors.Select(x => x.ErrorMessage).ToList();
+                return new ApiResponse<NewUserDto>(errorMessages: "Validation failed", errors: errorsValidation);
+            }
+
+            var existingUser = await _userManager.FindByEmailAsync(registerDto.Email);
+            if (existingUser != null)
+            {
+                return new ApiResponse<NewUserDto>(
+                    errorMessages: "Register failed",
+                    errors: new List<string> { "Email is already in use" }
+                );
+            }
+
+            var appUser = new AppUser
+            {
+                UserName = registerDto.UserName,
+                Email = registerDto.Email,
+            };
+
+            var createdUser = await _userManager.CreateAsync(appUser, registerDto.Password);
+
+            if (createdUser.Succeeded)
+            {
+                var roleResult = await _userManager.AddToRoleAsync(appUser, "User");
+                if (roleResult.Succeeded)
+                {
+                    var newUser = new NewUserDto()
+                    {
+                        Username = appUser.UserName,
+                        Email = appUser.Email,
+                        Token = _tokenService.CreateToken(appUser)
+                    };
+
+                    return new ApiResponse<NewUserDto>(newUser, "Success");
+                }
+                else
+                {
+                    var errorsRole = roleResult.Errors.Select(x => x.Description).ToList();
+                    return new ApiResponse<NewUserDto>(errorMessages: "Register failed", errors: errorsRole);
+                }
+            }
+            else
+            {
+                var errorsUser = createdUser.Errors.Select(x => x.Description).ToList();
+                return new ApiResponse<NewUserDto>(errorMessages: "Register failed", errors: errorsUser);
+            }
+
```

Comment:
Invert `if` blocks to reduce nesting
https://stackoverflow.com/questions/268132/why-should-i-invert-if-statement-to-reduce-nesting